snap/local/README.md

```diff
@@ -0,0 +1,411 @@
+
+To download files using DO Agent snap, the DU Agent snap need to connect to the following `slots` provided by the DO Agent snap:
+
+```yaml
+# Note: snippet from https://github.com/microsoft/do-client/blob/develop/snapcraft-options/snapcraft-agent.yaml
+
+slots:
+  do-port-numbers:
+    interface: content
+    content: do-port-numbers
+    read: [ $SNAP_DATA/run ]
+
+  do-configs:
+    interface: content
+    content: do-configs
+    write: [ $SNAP_DATA/etc ]
+
+plugs:
+  deviceupdate-agent-downloads:
+    interface: content
+    content: deviceupdate-agent-downloads
+    target: $SNAP_DATA/deviceupdate-agent-downloads
+
+layout:
+  # DU agent will provide the /var/lib path as download location.
+  # Map it to the correct path within this snap.
+  /var/lib/deviceupdate-agent-downloads:
+    symlink: $SNAP_DATA/deviceupdate-agent-downloads
+```
+
+You can list a DO client snap's connections by using  the `snap connections` command. This command displays information about the snap's connections, including its slots and plugs.
+
+```shell
+adu-dev@du-ubuntu-core-2004-build:~$ snap connections deliveryoptimization-client
+
+Interface     Plug                                                      Slot                                         Notes
+content       -                                                         deliveryoptimization-client:do-configs       -
+content       -                                                         deliveryoptimization-client:do-port-numbers  -
+content       deliveryoptimization-client:deviceupdate-agent-downloads  -                                            -
+network       deliveryoptimization-client:network                       :network                                     -
+network-bind  deliveryoptimization-client:network-bind                  :network-bind                                -
+
+```
+
+DU Agent snap will declare the following `plugs` and `layout`
+
+```yaml
+# Note: from ../snapcraft.yaml
+plugs:
+  do-port-numbers:
+    interface: content
+    content: do-port-numbers
+    target: $SNAP_DATA/do-port-numbers
+
+  do-configs:
+    interface: content
+    content: do-configs
+    target: $SNAP_DATA/do-configs
+
+  snapd-control:
+    interface: snapd-control
+
+# Provides resources to be accssed by other snaps.
+slots:
+  #
+  # A downloads folder.
+  #
+  # Consumer(s):
+  #
+  #    Delivery Optimization : writes downloaded file to this folder and/or sub-folders.
+  #
+  downloads-folder:
+    interface: content
+    content: downloads-folder
+    write:
+        - $SNAP_DATA/data/downloads
+
+layout:
+# adu_data_dir
+  /var/lib/adu:
+    symlink: $SNAP_DATA/data
+
+  /var/lib/deviceupdate-agent-downloads:
+    symlink: $SNAP_DATA/data/downloads
+
+# adu_conf_dir
+  /etc/adu:
+    symlink: $SNAP_DATA/config
+# adu_shell_dir
+  /usr/lib/adu:
+    symlink: $SNAP/usr/lib/adu
+# adu_log_dir
+  /var/log/adu:
+    symlink: $SNAP_DATA/log
+# curl command
+  /usr/bin/curl-downloader:
```

Comment:
DO downloader is mentioned above. Is this curl command still required?  Same for line 298

Reply:
Yes, still need this. In case we want to switch to curl downloader.